Combine 'read_argfile.py' and 'kernels.py' into 're_matrix.py'

HarelErlich opened the pull request.

HarelErlich committed:
Combine 'read_argfile.py' and 'kernels.py' into 're_matrix.py'

chr15p commented on scripts/re_matrix.py:
```diff
@@ -2,17 +2,45 @@
+        return f"Error download file: {e}"
+
+def fetch_dtk_tags():
+    dtk_registry_api = "https://quay.io/v1/repositories/build-and-sign/pa-driver-toolkit/tags"
```
not a fan of having a url hardcoded somewhere in the depths of a function, could we read it from argsfile.conf, or maybe make it a global variable so it sits at the top of the file whcih would at least make it obvious

HarelErlich replied:
Added this as a variable to `argsfile.conf`

chr15p commented on scripts/re_matrix.py:
```diff
@@ -2,17 +2,45 @@
+    #  Get DTK tags
+    response = requests.get(dtk_registry_api)
+    kernel_json = (response.json())
+    kernel_list = "data/kernel-list.json"
```
we have this filename hardcoded here AND hardcoded further down in the file so if we want to change it we have to do it in (at least) two places and if we miss one it will cause hard to debug issues, so we probably want it defined in only one place, so we cuold set it as a global variable, or pass it into this function as a parameter, or something else?  (or see the next comment :) )

chr15p commented on scripts/re_matrix.py:
```diff
@@ -2,17 +2,45 @@
+    kernel_list = "data/kernel-list.json"
+    # Save tags in file
+    with open(kernel_list, "w") as output:
+        json.dump(kernel_json, output)
```
we are writing kernel-list.json out here, then reading it back in a few lines later which feels unnecessary, maybe this function should pass back a data structure with the kernels and skip both filesystem operations

HarelErlich replied:
Removed the filesystem operations and returned it as a JSON object.

chr15p commented on scripts/re_matrix.py:
```diff
@@ -2,17 +2,45 @@
-import read_argfile
+import requests
+
+def read_key_value_file(filename="argfile.conf"):
```
I think check_changes.py also reads this file in with a very similar function so it probably makes sense to have a library with this (and other) duplicated functionality. Although maybe this is something for a separate PR when we understand what actually is duplicated

HarelErlich replied:
You're right, it reads it with another function, and I was looking for `import read_argfile` but couldn't find it.
I created a `read_argfile.py` with one function, and we can change it later.

chr15p commented on scripts/kernels.py:
this script is referenced in .github/workflows/create-matrix.yaml  so that will need removing from as well.

HarelErlich committed:
Refactor DTK API handling and improve config management

- Moved the DTK registry API URL to `argfile.conf` for better flexibility.
- Removed the need for `kernel-list.json` by directly pulling data from the DTK registry API.
- Created `read_argfile.py` as a shared module and used it in `re_matrix.py` and `check_changes.py`.

HarelErlich commented on argfile.conf:
I don't know if `DTK_REGISTRY_API` is a good name for this variable, but I followed the naming convention used before
